### ISSUE TITLE
Add delay before components drawer is opened

### ISSUE THREAD

**Issue by bytasv** [mui/mui-toolpad#1524]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

If I move mouse through the screen not fast enough I can see components drawer opening and closing. This can be particularly annoying when moving from the left side of the screen to the component which is then blocked behind drawer.

https://user-images.githubusercontent.com/437214/211000500-83c52f39-3928-4426-b19e-edf5b488d740.mp4

We could add a delay before drawer is opened so it's not opening/closing always just by moving mouse over it

**Comment by Janpot** [comment]
This was done deliberately. There is no way to distinguish between the two scenarios without introducing a delay that also applies when the intention is opening the drawer. To remedy the case of dragging past the drawer, we omit the closing delay so that the partially visible drawer disappears immediately.

**Comment by apedroferreira** [comment]
I agree with a noticeable delay not sounding like a great idea, but if it's short and it helps the UX feel less weird I guess it should be fine.
Also, because dragging components might be one of the main user interactions in Toolpad, I wonder if it should really be hidden in a drawer. I guess that in most low-code tools this panel is probably always visible. But of course it might not be a simple problem to find space for everything in the UI.

**Comment by bytasv** [comment]
> a noticeable delay

Just to clarify, noticeable delay or worse UX was not mentioned in the original description :) I think there are at least few things that could be tried out without degrading UX.

Main point of this issue is that drawer opening when I as user don't intend that to happen is not the best experience and if this can be improved, it would be great.

**Comment by prakhargupta1** [comment]
@gerdadesign Bringing this issue to your notice and adding it in the current cycle.

**Comment by prakhargupta1** [comment]
I think the drawer behavior is somewhat tricky to crack. Even after adding delay, it could be annoying to some. So maybe it makes sense to access the component library after a click. 

If we do it then keeping it as a drawer or a button in left sidebar/top navbar would be worth discussing.
